runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1347,6 +1356,80 @@ TR::Node *TR_VectorAPIExpansion::binaryIntrinsicHandler(TR_VectorAPIExpansion *o
    }
 
 
+
+TR::Node *TR_VectorAPIExpansion::broadcastCoercedIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
+                                                        TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+   {
+   TR::Compilation *comp = opt->comp();
+
+   if (mode == checkScalarization)
+      return node;
+
+   if (mode == checkVectorization)
+      {
+      if (!comp->cg()->getSupportsOpCodeForAutoSIMD(TR::vsplats, elementType))
+         return NULL;
+      else
+         return node;
+      }
+
+   if (opt->_trace)
+      traceMsg(comp, "broadcastCoercedIntrinsicHandler for node %p\n", node);
+
+   int32_t elementSize = OMR::DataType::getSize(elementType);
+   TR::Node *valueToBroadcast = node->getChild(3);
+
+   anchorOldChildren(opt, treeTop, node);
+
+   TR::Node *newNode;
+
+   switch (elementType) {
+      case TR::Float:
+          newNode = TR::Node::create(node, TR::ibits2f, 1, TR::Node::create(node, TR::l2i, 1, valueToBroadcast));
+          break;
+      case TR::Double:
+          newNode = TR::Node::create(node, TR::lbits2d, 1, valueToBroadcast);
+          break;
+      case TR::Int8:
+          newNode = TR::Node::create(node, TR::l2b, 1, valueToBroadcast);
+          break;
+      case TR::Int16:
+          newNode = TR::Node::create(node, TR::l2s, 1, valueToBroadcast);
+          break;
+      case TR::Int32:
+          newNode = TR::Node::create(node, TR::l2i, 1, valueToBroadcast);
+          break;
+      case TR::Int64:
+          newNode = TR::Node::create(node, TR::dbits2l, 1, TR::Node::create(node, TR::lbits2d, 1, valueToBroadcast));
```

Comment:
I suppose these redundant conversions are to make sure that the `doScalarization` path can easily recreate `node` as a copy of `newNode`. I would recommend `PassThrough` for that purpose instead, but it looks like its `getDataType()` will return `NoType`, which I don't think will work with vector opcode type inference - maybe we should change that

Anyway, it might be good to have a code comment to explain what's going on here

Reply:
Yes, that's correct: just to make node recreation easier. I will put a comment of course.

Reply:
added a comment